src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier color = new ColorSupplier();
+    private static final Random random = new Random();
+    private static final FigureList[] figures = FigureList.values();
+    private static final double RANDOM_MAX_VALUE = 5.1;
+
+    public Figure getRandomFigure() {
+        Figure figure = null;
+
+        switch (randomiseFigureType()) {
+            case CIRCLE:
+                figure = new Circle(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case ISOSCELES_TRAPEZOID:
+                figure = new IsoscelesTrapezoid(color.getRandomColor(),
+                        random.nextDouble(RANDOM_MAX_VALUE), random.nextDouble(RANDOM_MAX_VALUE),
+                        random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case RIGHT_TRIANGLE:
+                figure = new RightTriangle(color.getRandomColor(),
+                        random.nextDouble(RANDOM_MAX_VALUE), random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case RECTANGLE:
+                figure = new Rectangle(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE),
+                        random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case SQUARE:
+                figure = new Square(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            default:
+        }
+        return figure;
```

Comment:
Returning 'null' from a method is a bad practice. Instead of initializing 'figure' to null, you can initialize it directly in the 'default' case of the switch block. This way, you will never return null and you can also remove the 'figure' variable declaration at the beginning of the 'getRandomFigure()' method.